service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -106,13 +106,29 @@ public String getWsmServiceAccountToken() throws IOException {
    * @param userRequest - request object for this user
    * @return - email address of user
    */
-  public String getRequestUserEmail(AuthenticatedUserRequest userRequest)
+  public String getEmailFromRequestOrSam(AuthenticatedUserRequest userRequest)
       throws InterruptedException {
     Optional<String> emailMaybe = Optional.ofNullable(userRequest.getEmail());
     if (emailMaybe.isPresent()) {
       return emailMaybe.get();
     } else {
-      return getEmailFromToken(userRequest.getRequiredToken());
+      return getEmailFromSam(userRequest);
+    }
+  }
+
+  /**
+   * Fetch the email associated with user credentials directly from Sam. Unlike
+   * {@code getRequestUserEmail}, this will always call Sam to fetch an email and will never read
+   * it from the AuthenticatedUserRequest. This is important for calls made by pet service accounts,
+   * which will have a pet email in the AuthenticatedUserRequest, but Sam will return the owner's
+   * email.
+   * */
+  public String getEmailFromSam(AuthenticatedUserRequest userRequest) throws InterruptedException {
```

Comment:
maybe `getUserEmailFromSam()`. It kinda seems redundant on the `SamService` to have to say `fromSam` but I agree it's a necessary evil.